src/main/java/model/structures/GraphPathImpl.java

```diff
@@ -0,0 +1,126 @@
+package model.structures;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Random;
+import java.util.stream.Collectors;
+
+public class GraphPathImpl implements GraphPath {
+
+    private ArrayList<Node> path;
+    private int sumOfWages;
+    private ArrayList<Integer> wages;
+    private int[][] adjacencyMatrix;
+
+    public GraphPathImpl(int[][] adjacencyMatrix)
+    {
+        path = new ArrayList<>();
+        wages = new ArrayList<>();
+        sumOfWages = 0;
+        this.adjacencyMatrix = adjacencyMatrix;
+    }
+
+    private GraphPathImpl(int[][] adjacencyMatrix, int sumOfWages, ArrayList<Integer> wages, ArrayList<Node> path)
+    {
+        this.adjacencyMatrix = adjacencyMatrix;
+        this.sumOfWages = sumOfWages;
+        this.wages = wages;
+        this.path = path;
+    }
+
+    public void addToPath(Node node, int wage)
+    {
+        Node theNewNode = new Node(node);
+
+        path.add(theNewNode);
+        if(wage != 0)
+            wages.add(wage);
+        sumOfWages += wage;
+    }
+
+    public void addToPath(Node node)
+    {
+        addToPath(node, 0);
+    }
+
+    public ArrayList<Node> getPath() {
```

Comment:
Ta metoda zle sie nazywa. Gdzie jest uzywana?

Reply:
To jest getter do zmiennej Path, jeśli nie potrzebujesz to go usunę (zakładałem, że użyjesz tego w symulacji)